services/api/tests/test_codex_runtime.py

```
@@ -0,0 +1,170 @@
+        encoding="utf-8",
+    )
+    server.chmod(0o700)
+    client = CodexAppServerProcess(
+        server,
+        tmp_path / "codex-home",
+        request_timeout=0.25,
+        verify_binary=lambda _: None,
+    )
+    try:
+        await client.start()
+        with pytest.raises(CodexRuntimeError, match="timed out|stopped"):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)